lib/web_ui/dev/test_platform.dart

```diff
@@ -141,8 +144,9 @@ class BrowserPlatform extends PlatformPlugin {
   }
 
   Future<shelf.Response> _screeshotHandler(shelf.Request request) async {
-    if (browserName != 'chrome') {
-      throw Exception('Screenshots tests are only available in Chrome.');
+    if (browserName != 'chrome' && browserName != 'ios-safari') {
```

Comment:
It looks like there are multiple places where we check for supported browsers. What if we had a method on `ScreenshotManager` to help with this? Something like this:
```dart
typedef ScreenshotManagerFactory = ScreenshotManager Function();

abstract class ScreenshotManager {
  final Map<String, ScreenshotManagerFactory> _browserFactories = <String, ScreenshotManagerFactory>{
    kBrowserChrome: () => ChromeScreenshotManager(),
    kBrowserIOSSafari: () => IOSSafariScreenshotManager(),
  };

  bool isBrowserSupported(String browser) => _browserFactories.containsKey(browser);

  static ScreenshotManager choose(String browser) {
    if (isBrowserSupported(browser)) {
      return _browserFactories[browser]();
    }
    throw StateError('...');
  }
}
```

Reply:
done. thanks a lot! this is really nice.